[RNMobile] Image Size improvements mark I

etoledom opened the pull request:
This PR fixes 4 and 5 from https://github.com/wordpress-mobile/gutenberg-mobile/issues/1593

On web, then the selected image object is received, it already has all the information needed (image sizes details) to set the proper url for the default slug. But in our case, we only get the original image URL.

For this reason it was needed to use `componentDidUpdate` to capture the new `image` object from props when it is set and update the image url.

One step-back of this approach is that we will load two images, the original one, and the large one when we get the Image object.

`gutenberg-mobile` side PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1598
Related `WPiOS` PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/12984

**To test:**

- On WPiOS or WPAndroid.
- IF WPiOS: Checkout https://github.com/wordpress-mobile/WordPress-iOS/pull/12984
- Set a new Image block.
- Set add an image to the block (any source should work).
- Switch to HTML mode.
  - IF WPCom: Check that the image URL has the `?w=xxx` property.
  - IF Self-Hosted: Check that the image URL is the proper resized variant: `...img_2782-768x1024.jpg` or similar.

etoledom committed:
Setting image size url for default slug, and apply the default when image changes.

mchowning commented on packages/block-library/src/image/edit.native.js:
```diff
@@ -309,7 +343,7 @@ export class ImageEdit extends React.Component {
 							icon={ 'editor-expand' }
 							label={ __( 'Size' ) }
-							value={ sizeOptionLabels[ sizeSlug || DEFAULT_SIZE_SLUG ] }
+							value={ sizeSlug }
```
Not using the `sizeOptionLabels` here is causing the active size slug to display lower case (i.e. "large") in the UI. I think we want to still use the labels here.

![Screen Shot 2019-11-20 at 11 43 38 AM](https://user-images.githubusercontent.com/4656348/69258687-00342b00-0b8b-11ea-95ad-3f8bffce4fd4.png)

etoledom replied:
You are completely right. I'll revert this change.

mchowning commented:
I am seeing an issue where it looks like iOS isn't getting the image url for my pressable site. iOS has a url like:

```
...com/wp-content/uploads/2019/11/wp-15741816431884373934719785215724-scaled.jpg
```

but Android has this url:
```
...com/wp-content/uploads/2019/11/wp-15741816431884373934719785215724-768x1024.jpg
```

In addition to the url difference, I cannot update the size of the image on iOS only (probably because the urls are not being fetched). I see the same issue if I take a photo on the device with iOS--the url still has "-scaled" and I cannot change the size. I can create a post on Android with that same image (originating from the iOS device) and it gets the proper url and can be resized.

I'm seeing this issue on both the WPiOS `develop` (75a83debc4) and `gutenberg/image-size-improvements` (2638c756d5) branches.

## Full blocks

### iOS
```
<!-- wp:image {"id":4,"sizeSlug":"large"} -->
  <figure class="wp-block-image size-large">
    <img src="https://mattcpressable.mystagingwebsite.com/wp-content/uploads/2019/11/wp-15741816431884373934719785215724-scaled.jpg" alt="" class="wp-image-4"/>
  </figure>
<!-- /wp:image -->
```
### Android
```
<!-- wp:image {"id":4,"sizeSlug":"large"} -->
  <figure class="wp-block-image size-large">
    <img src="https://mattcpressable.mystagingwebsite.com/wp-content/uploads/2019/11/wp-15741816431884373934719785215724-768x1024.jpg" alt="" class="wp-image-4"/>
  </figure>
<!-- /wp:image -->
```

etoledom committed:
Using user facing SizeSlug label to show to the user.

etoledom commented:
> I am seeing an issue where it looks like iOS isn't getting the image url for my pressable site.

Good catch! This turned out to be an issue in the native side. I have fixed it on this PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/12984

Checking out the branch from that PR should be enough to give this PR another try (running metro server).

Updated the description accordingly 👍

mchowning commented:
Found one more issue, but it is independent of the changes in this PR, so I'll add this to our master issue for this.

In particular, the issue that I found is that (for some reason) one of the images on my pressable site only has the sizes "thumbnail", "medium", "medium_large", and "full". Note no "large" size. As a result, when the `getUrlForSlug` method was called with a "large" `sizeSlug` it returns undefined. The result of this is that on selecting this image from my media library it would briefly display in the image block, but then quickly flicker away and return to the "Add Image" state after `getUrlForSlug` tries (unsuccessfully) to get a url for the default "large" size.

mchowning reviewed (approved):
Looks good apart from the unexpected-sizeSlug issue I just found. That issue does not involve these changes though, so I added it to the master issue, and I think this is ready to go! 👍

mchowning commented on packages/block-library/src/image/edit.native.js:
```diff
@@ -71,6 +71,10 @@ const sizeOptions = map( sizeOptionLabels, ( label, option ) => ( { value: optio
 // Default Image ratio 4:3
 const IMAGE_ASPECT_RATIO = 4 / 3;
 
+const getUrlForSlug = ( image, { sizeSlug } ) => {
```
Just curious if there was a specific reason you chose to pass an `attributes`-like object with a `sizeSlug` key here instead of just passing the `sizeSlug` directly.

etoledom replied:
Basically just to simplify the call here:
```javascript
 const { image, attributes } = this.props; 	
 const url = getUrlForSlug( image, attributes );
```

But I'm not very happy about it since it also makes this other call a bit more complicated:
```javascript
const url = getUrlForSlug( image, { sizeSlug } );
```

etoledom committed:
Merge remote-tracking branch 'origin/master' into rnmobile/image-size-improvements
